src/coreclr/System.Private.CoreLib/src/System/GC.CoreCLR.cs

```diff
@@ -737,5 +737,53 @@ public static TimeSpan GetTotalPauseDuration()
         {
             return new TimeSpan(_GetTotalPauseDuration());
         }
+
+        internal delegate void EnumerateConfigurationValuesCallback(int type, long intValue, bool boolValue);
+
+        /// <summary>
+        /// Returns a dictionary keyed on GC Configuration Names with values as the configuration values.
+        /// </summary>
+        /// <remarks>
+        /// Corresponding issue: https://github.com/dotnet/runtime/issues/67471
+        /// </remarks>
+        public static unsafe IReadOnlyDictionary<string, object> GetConfigurationVariables()
+        {
+            Dictionary<string, object> result = new Dictionary<string, object>();
+
+            string? name  = null;
+            string? value = null;
+
+            // It is important that we do not capture name or value in the lambda
+            // That capture will lead to the failure of QCall::StringHandleOnStack.Set()
+            // because name is no longer allocated on the stack.
+            void* addressOfName  = (void*)Unsafe.AsPointer(ref name);
+            void* addressOfValue = (void*)Unsafe.AsPointer(ref value);
+            StringHandleOnStack nameHandle  = new StringHandleOnStack(ref name);
+            StringHandleOnStack valueHandle = new StringHandleOnStack(ref value);
```

Comment:
This should not be using StringHandleOnStack. Instead, the string should be passed to the callback as unmanaged pointer and the managed string object allocated in managed code. Just like you would do it if you were writing this in regular library outside corelib.

Reply:
In case this isn't clear how to do this, the idea is very simple and reduces the complexity of the lifetimes.

```csharp
[UnmanagedCallersOnly]
private static void Callback(void* strUtf8Raw)
{
    string strUtf8 = Marshal.PtrToStringUTF8((IntPtr)strUtf8Raw);
    ...
}

// Pass to native via address-of
CallIntoRuntime(&Callback);
```